.github/workflows/build-and-publish-release.yml

```diff
@@ -57,12 +57,11 @@ jobs:
         id: branch
         run: |
           VERSION_NUMBER=${{ steps.version_number.outputs.RELEASE_VERSION }}
-          echo "BRANCH_NAME=v${VERSION_NUMBER%.*}" >> $GITHUB_ENV
+          echo "BRANCH_NAME=v${VERSION_NUMBER%.*}" >> $GITHUB_OUTPUT
```

Comment:
where do you reference BRANCH_NAME?

Reply:
in checkout repo right below this. 

The intent and i can add comments here is that you push tag v1.111.0 for example:

1. We make RELEASE_VERSION lose the v for reference later.
2. We do add v back and lose the last .x (patch number) and use this for branch name.
3. Checkout branch name so we are on the right branch when we build.
4. Image tag uses RELEASE_VERSION (dropped v, and full release number so in this case 1.111.0)